File: JET/modul_1_c#_basics/L04KlassenMethoden/L04KlassenMethoden/Klassen/src/Pudel.cs
namespace Hunde;

public class Pudel : Hund // Ist-Beziehungen
{
    // private Felder
    private double fluff;

    // Konstruktor
    public Pudel(string name, int alter, string geschlecht, Double health, bool chipped, double fluff) 
        : base(name, alter, geschlecht, health, chipped)
    {
        this.fluff = fluff;
    }

    public Pudel(
        string name, int alter, string geschlecht, Double health, bool chipped, 
        double fluff, HundeBesitzer besitzer, Hund spielFreund
    ) 
        : this(name, alter, geschlecht, health, chipped, fluff)
    {
        this.SetBesitzer(besitzer);
        this.SetSpielFreund(spielFreund);
    }

    public Pudel(
        string name, int alter, string geschlecht, Double health, 
        bool chipped, double fluff, HundeBesitzer besitzer
    ) 
        : this(name, alter, geschlecht, health, chipped, fluff)
    {
        
        this.SetBesitzer(besitzer);
    }

    public Pudel(
        string name, int alter, string geschlecht, Double health, bool chipped, double fluff, Hund spielFreund
    )
        : this(name, alter, geschlecht, health, chipped, fluff)
    {


        this.SetSpielFreund(spielFreund);
    }

    // Methoden
    public void Winseln()
    {
        Console.WriteLine(".... ewww");
        SetHealth(GetHealth() - 1);
    }

    // Get-und-Set-Methoden
    public double GetFluff()
    {
        return fluff;
    }

    public void SetFluff(double fluff)
    {
        this.fluff = fluff;
    }
}

File: BITEC/modul_2_object_orientation/src/lerneinheiten/L02KlassenUndMethoden/einfuehrendes_beispiel/Pudel.java
package lerneinheiten.L02KlassenUndMethoden.einfuehrendes_beispiel;

public class Pudel extends Hund {
    double fluff;

    public Pudel(String name, Integer alter, String geschlecht, Double health, boolean chipped, double fluff) {
        super(name, alter, geschlecht, health, chipped);
        this.fluff = fluff;
    }

    public Pudel(String name, Integer alter, String geschlecht, Double health, boolean chipped, double fluff, HundeBesitzer besitzer, Hund spielFreund) {
        this(name, alter, geschlecht, health, chipped, fluff);
        this.setBesitzer(besitzer);
        this.setSpielFreund(spielFreund);
    }

    public Pudel(String name, Integer alter, String geschlecht, Double health, boolean chipped, double fluff, HundeBesitzer besitzer) {
        this(name, alter, geschlecht, health, chipped, fluff);
        this.setBesitzer(besitzer);
    }

    public Pudel(String name, Integer alter, String geschlecht, Double health, boolean chipped, double fluff, Hund spielFreund) {
        this(name, alter, geschlecht, health, chipped, fluff);
        this.setSpielFreund(spielFreund);
    }

    public void winseln() {
        System.out.println(".... ewww");
        setHealth(getHealth() - 1);
    }

    public double getFluff() {
        return fluff;
    }

    public void setFluff(double fluff) {
        this.fluff = fluff;
    }
}
